src/go/k8s/pkg/networking/ports.go

```diff
@@ -0,0 +1,132 @@
+// Package networking defines common networking logic for redpanda clusters
+package networking
+
+import (
+	redpandav1alpha1 "github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	"github.com/vectorizedio/redpanda/src/go/k8s/pkg/resources"
+)
+
+// PortsDefinition defines internal/external pair of ports for exposed services
+type PortsDefinition struct {
+	Internal *resources.NamedServicePort
```

Comment:
I don't like that we limit each API to only have 1 internal listener and 1 external listener.
If Redpanda supports multiple listeners and if we already receive an array of listeners in the CR, why limiting ourselves in here?

Reply:
because in the beginning we had this rule, then we added support for any number of listeners to the api but kept this rule in place to not make the PR too big. Well and then we never followed up to allow any number of listeners and there was no ask from anyone to do this :D That's the whole story

Reply:
Shouldn't we stop limiting to only 2 listeners and instead start supporting multiple listeners wherever possible, like in this PR?

Reply:
I agree with Alena, out of scope for this PR.